### ISSUE TITLE
Evaluation Metric？

### ISSUE THREAD

**Issue by Mortal825** [intelpro/ELEDNet#10]
Thank you for your excellent work. When reproducing your code, I noticed that the **test_model.py** lacks the code for calculating the PSNR and SSIM evaluation metrics. After I supplemented this part, I found that the results are inconsistent with those in the paper. I wonder if you could provide the complete **test_model.py**.

| ELEDNet             | PSNR  | SSIM   |
| ------------------- | ----- | ------ |
| small (weights provided by the author) | 29.59 | 0.8853 |
| large (self-trained)   | 31.17 | 0.9089 |
Inconsistent with the paper
<img width="796" height="115" alt="Image" src="https://github.com/user-attachments/assets/4846e81e-5d9f-46c0-876a-b8e729dc29e6" />

**Comment by intelpro** [comment]
```
def ssim_calculate(img1, img2, sd=1.5, C1=0.01**2, C2=0.03**2):
    # Processing input image
    img1 = np.array(img1, dtype=np.float32) / 255
    img1 = img1.transpose((2, 0, 1))

    # Processing gt image
    img2 = np.array(img2, dtype=np.float32) / 255
    img2 = img2.transpose((2, 0, 1))


    mu1 = gaussian_filter(img1, sd)
    mu2 = gaussian_filter(img2, sd)
    mu1_sq = mu1 * mu1
    mu2_sq = mu2 * mu2
    mu1_mu2 = mu1 * mu2
    sigma1_sq = gaussian_filter(img1 * img1, sd) - mu1_sq
    sigma2_sq = gaussian_filter(img2 * img2, sd) - mu2_sq
    sigma12 = gaussian_filter(img1 * img2, sd) - mu1_mu2

    ssim_num = ((2 * mu1_mu2 + C1) * (2 * sigma12 + C2))

    ssim_den = ((mu1_sq + mu2_sq + C1) * (sigma1_sq + sigma2_sq + C2))

    ssim_map = ssim_num / ssim_den
    return np.mean(ssim_map)
```

```
def psnr_calculate(x, y, val_range=255.0):
    x = x.astype(np.float)
    y = y.astype(np.float)
    diff = (x - y) / val_range
    mse = np.mean(diff ** 2)
    psnr = -10 * np.log10(mse)
    return psnr
```

**Comment by intelpro** [comment]
Please try this code. I’ve double-checked the evaluation results of the pretrained model and confirmed there are no issues. Thank you.
